luxonis_ml/data/datasets/luxonis_dataset.py

```diff
@@ -503,9 +503,38 @@ def add(self, generator: DatasetIterator, batch_size: int = 1_000_000) -> Self:
 
         self.fs.put_file(tmp_file.name, "metadata/file_index.parquet")
         self._write_metadata()
-
+        self.warn_on_duplicates()
         return self
 
+    def warn_on_duplicates(self) -> None:
+        df = self._load_df_offline()
+
+        # Warn on duplicate UUIDs
+        duplicates_paired = (
+            df.group_by("uuid")
+            .agg(pl.col("file").n_unique().alias("file_count"))
+            .filter(pl.col("file_count") > 1)
+            .join(df, on="uuid")
+            .select(["uuid", "file"])
+            .unique()
+            .group_by("uuid")
+            .agg([pl.col("file").alias("files")])
```

Comment:
Could this be further extended with `pl.len` and a filter for len > 1 so we don't have to do it in `iterrows`? It should make it faster. 
Not super important since this shouldn't be a bottleneck, but I'd assume issues like this are more likely the larger the dataset is, so it would be good to have this optimized.

Reply:
Made changes in the new commit